ludwig/modules/tabnet_modules.py

```diff
@@ -182,10 +182,11 @@ def __init__(
         self.size = size
         units = size * 2 if apply_glu else size
 
-        if shared_fc_layer:
+        # Initialize fc_layer before assigning to shared layer for torchscript compatibilty
+        self.fc_layer = nn.Linear(input_size, units, bias=False)
+        if shared_fc_layer is not None:
+            assert shared_fc_layer.weight.shape == self.fc_layer.weight.shape
```

Comment:
Is this a programming error if it happens, or a user error in the config? Want to make sure we aren't asserting on user errors, as those cannot be recovered from (and are generally disabled at runtime).

Reply:
IIUC is a programming error– this should essentially never happen unless something was implemented incorrectly